src/sidebar/store/test/index-test.js

```diff
@@ -148,3 +152,45 @@ describe('createSidebarStore', () => {
     });
   });
 });
+
+describe('useSidebarStore', () => {
+  function AnnotationCard({ id }) {
+    const store = useSidebarStore();
+    const ann = store.findAnnotationByID(id);
+    return <div>{ann.text}</div>;
+  }
+
+  // `useSidebarStore` is a trivial wrapper, so rather than mock its dependencies,
+  // this is a more useful integration test that covers interaction of the store
+  // and UI components.
+  it('returns wrapper for components to interact with store', () => {
+    const store = createSidebarStore({});
+    const annot = { ...defaultAnnotation(), text: 'Initial text' };
+    store.addAnnotations([annot]);
+
+    const services = {
+      get(service) {
+        return service === 'store' ? store : null;
+      },
+    };
+
+    const el = document.createElement('div');
+    act(() => {
+      render(
+        <ServiceContext.Provider value={services}>
+          <AnnotationCard id={annot.id} />
+        </ServiceContext.Provider>,
+        el
+      );
+    });
+    assert.equal(el.innerHTML, '<div>Initial text</div>');
+
+    act(() => {
+      const updatedAnnot = { ...annot, text: 'Updated text' };
+      store.addAnnotations([updatedAnnot]);
+    });
+    assert.equal(el.innerHTML, '<div>Updated text</div>');
+
+    render(null, el); // Force unmount and cleanup of subscribers
```

Comment:
Curious: Is this something you remembered to do naturally or were there side-effects that prompted it? I feel like this is something I could have very easily overlooked my-own-self.

Reply:
The `useSidebarStore` hook has to be creating some kind of subscription, regardless of implementation, so I wanted to force that to be cleaned up.